### PR TITLE
[Merge into #16] [Feature] Checkout Tracking

### PR DESCRIPTION
**Pull request by MichaelCampbell**
Created `CheckoutTracking` protocol to describe checkout events.

### CODE REVIEW

File: Simcoe/mParticle/MPCommerceEventKeys.swift [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ - screenName:        The screen name.
+ - nonInteractive:    The non interactive flag.
+ */
+internal enum MPCommerceEventKeys: String {
+
+    case currency = "currency"
+    case productListName = "productListName"
+    case productListSource = "productListSource"
+    case screenName = "screenName"
+    case nonInteractive = "nonInteractive"
```
Comment (htinlinn) [inline Simcoe/mParticle/MPCommerceEventKeys.swift]: CamelCase here too.

File: Simcoe/mParticle/mParticle.swift [hunk 2]
```diff
@@ -38,32 +38,74 @@ public class mParticle {
...
             product.position = position
+            additionalProperties?.removeValueForKey(MPProductKeys.position.rawValue)
         }
 
         return product
     }
 
+    private func mPCommerceEvent(eventType: MPCommerceEventAction,
+                                 product: MPProduct,
+                                 inout additionalProperties: Properties?) -> MPCommerceEvent {
```
Comment (htinlinn) [inline Simcoe/mParticle/mParticle.swift]: Would it make sense to create a separate dictionary for the purpose of passing SDK-specific properties?

**Comment by htinlinn** [comment]
I think we should either move the functions that are in `MPEvent+Simcoe.swift` into `mParticle.swift` or create two new `MPProduct+Simcoe.swift` and ``MPCommerceEvent+Simcoe.swift` files for the new functions.

File: Simcoe/mParticle/MPCommerceEventKeys.swift (hunk 1 above)
Comment (MichaelCampbell) [reply Simcoe/mParticle/MPCommerceEventKeys.swift]: - [x] CamelCase

File: Simcoe/mParticle/mParticle.swift (hunk 2 above)
Comment (MichaelCampbell) [reply Simcoe/mParticle/mParticle.swift]: How do you imagine that would work?

Comment (MichaelCampbell) [reply Simcoe/mParticle/mParticle.swift]: updated, please take another look

File: Simcoe/mParticle/MPTransactionAttributesKeys.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+//  Created by Michael Campbell on 11/2/16.
+//  Copyright © 2016 Prolific Interactive. All rights reserved.
+//
+
+/// The MPTransactionAttributes keys available.
+///
+/// - Affiliation: The affiliation.
+/// - CouponCode: The coupon code.
+/// - Revenue: The revenue amount.
+/// - Shipping: THe shipping amount.
```
Comment (yoyoyoseob) [inline Simcoe/mParticle/MPTransactionAttributesKeys.swift]: Small typo

Comment (MichaelCampbell) [reply Simcoe/mParticle/MPTransactionAttributesKeys.swift]: - [x] fixed typo

**Review by yoyoyoseob — APPROVED** [review]

**Pull request merged**